src/utils/api/sendJUP.ts

```diff
@@ -64,47 +16,57 @@ export interface ISignedTransaction extends IUnsignedTransaction {
 //   transactionBytes=001046aac6013c0057fb6f3a958e320bb49c4e81b4c2cf28b9f25d086c143
 
 // sendTransaction call to the API requires adminPassword so that cannot be used
-function sendJUP(unsigned: IUnsignedTransaction) {
-  let signedTx: ISignedTransaction | undefined;
+async function sendJUP(unsigned: IUnsignedTransaction) {
+  let signedTx: ISignedTransaction;
   let isValid: boolean;
   try {
     // sign
-    console.log("signing tx...", unsigned);
-    signedTx = signTx(unsigned);
+    signedTx = await signTx(unsigned);
     // validate
     isValid = validateTx(signedTx);
-    // send
+    console.log("isValid:", isValid);
+    // // broadcast
     if (isValid) {
-      console.log("valid transaction, broadcasting not implemented yet. signedTx:", signedTx, "isValid:", isValid);
-      return;
+      const broadcastResult = await broadcastTx(signedTx);
+      broadcastResult ? true : false;
     }
-    console.error("transaction invalid");
-    return;
+    return false;
   } catch (e) {
     console.error("error sendJUP():", e);
+    return false;
   }
 }
 
 //
 // Helper functions
 //
 
-function signTx(unsigned: IUnsignedTransaction) {
+async function signTx(unsigned: IUnsignedTransaction) {
   const secret = "test"; // TODO: implement
 
-  let result;
+  console.log("preparing to sign JSON:", unsigned);
+
+  let result: ISignedTransactionResult;
   try {
-    // result = API(BASEURL + "requestType=signTransaction&unsignedTransactionBytes=" + unsigned + "&secretPhrase=" + secret, "GET");
+    result = await API("requestType=signTransaction&unsignedTransactionJSON=" + JSON.stringify(unsigned) + "&secretPhrase=" + secret, "GET");
```

Comment:
would be smart to put url encoding of the parameters into the API code. it seems to me that json may not always contain characters that are friendly (like spaces) in a GET request without encoding for a URI parameter.

Reply:
added TODO for now since I want to discuss with you a bit more before taking an approach